### DISCUSSION TITLE
Even Polynomial Class

### DISCUSSION THREAD

**Discussion by pribossek** [mjhoptics/ray-optics#58] [General]
Dear Michael,

thank you very much for your efforts writing this library.
So far I am trying to use it for simple spherical lens systems in sequential mode and was impressed by the library capabilities.
I am still finding my way around the lib though. I would like to add some aspheric coefficents to some of the surfaces. May I ask how to define them? add_surface method does not seem to have this options, I noticed the EvenPolynomial Class but unfortunately did not find much information how to use it.
Is there a way to import the sag in discrete form numerically, does rayoptics provide methods to perform EvenPolynomial fits?

Regards,
Jaka

**Comment by pribossek** [comment]
I just stumbled across this notebook, maybe it helps others too: 
https://github.com/mjhoptics/ray-optics-notebooks/blob/master/Cell%20Phone%20lens.ipynb

**Reply by pribossek** [reply]
Is there a simple way to combine two toroid surfaces together?
I tried summing up the Xtoroid and Ytoroid profiles, but it did not work. 
aspx=XToroid(...)
aspy=YToroid(...)
sm.ifcs[2].profile=aspx+aspy

**Reply by mjhoptics** [reply]
Hello Jaka,
There isn't a "simple" way to combine surfaces like that (it would be nice though!).
At a minimum, a `SurfaceProfile` subclass needs to provide implementations of the `f()` and `df()` methods; these compute the function value and gradient for an input point. Look at the method definitions in `SurfaceProfile` and note the methods that are implemented using `pass`.
Regards,
Mike
